### PR TITLE
introduce NODEJS_ORG_MIRROR and IOJS_ORG_MIRROR

### PR DESCRIPTION
**Pull request by rvagg**
/R= @nodejs/build @ljharb

to be followed soon by a semver-major that removes the `NVM_` variants as these cause grief, see #787

### CODE REVIEW

**Comment by ljharb** [comment]
:+1:

**Comment by mgol** [comment]
I can't judge the build & tests but source changes look good. :+1:

**Comment by ralphtheninja** [comment]
LGTM

**Comment by rvagg** [comment]
Should we do a `warn` message (npmlog style) whenever `NVM_*` is encountered? I can't recall the details of how this is used internally in nvm but would it hit many users not trying to use those variables to control node-gyp? @ljharb?

**Comment by ralphtheninja** [comment]
> Should we do a warn message (npmlog style) whenever NVM_* is encountered?

Like a deprecated warning? +1

**Comment by ljharb** [comment]
@rvagg it's primarily that it's an env var that while I don't intend to change, I might - a warning seems reasonable so people know to switch.

**Comment by rvagg** [comment]
Added a warning, run the tests to see it output:

```
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
```

@bnoordhuis can I ask for a review here as well please?

@zkat @iarna are there any guidelines on outputting warnings with `npmlog`? Is this wording going to be OK when it starts showing up for users who have that env var set?

File: test/test-process-release.js [hunk 1]
```diff
@@ -519,3 +519,119 @@ test('test process release - process.release ~ node@4.0.0-rc.4 - bogus string pa
...
+})
+
+test('test process release - IOJS_ORG_MIRROR', function (t) {
+  t.plan(2)
+
+  process.env.IOJS_ORG_MIRROR = 'http://foo.bar'
+
+  var release = processRelease([], { opts: {} }, 'v3.2.24', {
+    name: 'io.js',
+    headersUrl: 'http://foo.bar/v3.2.24/iojs-v3.2.24-headers.tar.gz'
```
Comment (bnoordhuis) [inline test/test-process-release.js]: You use 'https://nodejs.org/` in the first two tests but `http://foo.bar` here and in the next test.  Is that intentional?

**Comment by bnoordhuis** [comment]
JS changes LGTM with a question, no opinion on the docker shell script.  There are some long lines but that's not really a thing with node-gyp.

File: test/test-process-release.js (hunk 1 above)
Comment (rvagg) [inline test/test-process-release.js]: No, search/replace error, the test should find that `process.release` is mostly ignored and it's not testing that as it is now. Thanks for noticing, will fix!

**Pull request closed without merging**